### PR TITLE
take feedback comment

### PR DESCRIPTION
**Pull request by kotarella1110**
take comments https://github.com/react-hook-form/resolvers/pull/32

### CODE REVIEW

File: src/yup.ts [hunk 1]
```diff
@@ -1,44 +1,46 @@
...
+                  ...(validateAllFieldCriteria
+                    ? {
+                        types: {
+                          ...((previous[path] && previous[path].types) || {}),
+                          [type]:
+                            previous[path] &&
+                            previous[path].types &&
+                            previous[path].types[type]
+                              ? [
+                                  ...[].concat(previous[path].types[type]),
```
Comment (bluebill1049) [inline src/yup.ts]: can we just use spread operator here?

**Review by bluebill1049 — APPROVED** [review]
👍

File: src/yup.ts (hunk 1 above)
Comment (kotarella1110) [reply src/yup.ts]: `previous[path].types[type]` type is `string | string[]`.
I think `[].concat(previous[path].types[type])` is better than `typeof previous[path].types[type] === 'string' ? [previous[path].types[type]] : previous[path].types[type]` 👍

Comment (bluebill1049) [reply src/yup.ts]: i thought `[...previous[path].types[type]]` is the same as `...[].concat(previous[path].types[type])`?

File: src/yup.test.ts [hunk 2]
```diff
@@ -140,21 +132,9 @@ describe('yupResolver', () => {
...
-        }
-      `);
-      expect(resolve.errors['password'].types).toMatchInlineSnapshot(`
-        Object {
-          "required": "password is a required field",
-        }
-      `);
+      expect(resolve.errors.age.types).toMatchInlineSnapshot(`undefined`);
+      expect(resolve.errors.createdOn.types).toMatchInlineSnapshot(`undefined`);
+      expect(resolve.errors.password.types).toMatchInlineSnapshot(`undefined`);
```
Comment (Paul-Vandell) [inline src/yup.test.ts]: Just for 3 of them `expect(resolve.errors.password.types).toBeUndefined()` finally.

File: src/yup.ts [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
+            ? {
+                [path]: {
+                  ...(previous[path] || {
+                    message,
+                    type,
+                  }),
+                  ...(validateAllFieldCriteria
+                    ? {
+                        types: {
+                          ...((previous[path] && previous[path].types) || {}),
```
Comment (Paul-Vandell) [inline src/yup.ts]: store in variable to avoid redundancy maybe with line 22 and 24. Not mandatory.

File: src/yup.ts (hunk 1 above)
Comment (Paul-Vandell) [reply src/yup.ts]: I propose this instead `[...(previous[path].types[type] || []), message]` to stay consistent with your code.

File: src/yup.ts (hunk 3 above)
Comment (Paul-Vandell) [reply src/yup.ts]: Not mandatory too to save few bytes as @bluebill tell me before

File: src/yup.ts (hunk 1 above)
Comment (kotarella1110) [reply src/yup.ts]: @bluebill1049 @Vandell63  umm... The code doesn't support when `previous[path].types[type]` is a string.
I think it's correct to use `concat`.

スクリーンショット 2020-07-21 17 23 14

```ts
const string1 = 'foo';
const string2 = 'bar';
// string
const test1 = [...string1, string2]; // ['f', 'o', 'o', 'bar']
const test2 = [...[].concat(string1), string2]; // ['foo', 'bar']
// string[]
const test3 = [...test2, string1]; // ['foo', 'bar', 'foo']
const test4 = [...[].concat(test2), string2]; // ['foo', 'bar', 'foo']
```

File: src/yup.test.ts (hunk 2 above)
Comment (kotarella1110) [reply src/yup.test.ts]: I will fix this 👍

File: src/yup.ts (hunk 3 above)
Comment (kotarella1110) [reply src/yup.ts:22]: Thanks. I will fix 👍

File: src/yup.ts (hunk 1 above)
Comment (bluebill1049) [reply src/yup.ts]: i see that's fine by me 👍

Comment (bluebill1049) [reply src/yup.ts]: thanks for the explanation.

File: src/yup.test.ts (hunk 2 above)
Comment (kotarella1110) [reply src/yup.test.ts]: done.

File: src/yup.ts (hunk 3 above)
Comment (kotarella1110) [reply src/yup.ts]: done.

**Review by Paul-Vandell — APPROVED** [review]

**Comment by bluebill1049** [comment]
thanks for the review @Vandell63 please merge it in 👍

**Pull request merged**